webapp/lib/controller.dart

```diff
@@ -90,6 +96,78 @@ void initUI() {
   );
 }
 
+Map<String, model.NeedsReplyData> getLatestDataForProjects(List<model.NeedsReplyData> updatedData) {
+  Map<String, model.NeedsReplyData> latestProjectData = {};
+  
+  for (var project in projectList) {
+    var data = updatedData.where((data) => data.project == project).toList();
+    
+    if (data.isEmpty) {
+      latestProjectData[project] = null;
+    } else {
+      latestProjectData[project] = data.last;
+    }
+  }
+  return latestProjectData;
+}
+
+bool isProjectStale(model.NeedsReplyData projectData) {
+  var now = new DateTime.now();
+  int lastUpdateTimeDiff =  now.difference(projectData.datetime).inHours;
+
+  if (lastUpdateTimeDiff >= 1) {
+      return true;
+    } else {
+      return false;
+    }
+}
+
+void setupProjectTimer(model.NeedsReplyData projectData, [bool stale = false]) {
+  for (var project in projectTimers.keys) {
+
+    if (projectTimers[project] != null) {
+      projectTimers[project].cancel();
+    }
+  }
+
```

Comment:
I don't think we should be cancelling all project timers.. shouldn't it only be the timer for the project given by `projectData`?

Reply:
Yes, actually it is. Fixed